### PR TITLE
Entry on the BEAM Search algorithm

### PR DESCRIPTION
**Pull request by HishamT**
### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [ ] All writings are my own.
- [ ] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [ ] I have performed a self-review of my own writing and code.
- [ ] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [ ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by KTom101 — CHANGES_REQUESTED** [review]
@HishamT I've added very minor comments.

File: content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+Title: 'BEAM Search'
+Description: 'A graph searching algorithm that is a variant of breadth-first search'
+Subjects:
+  - 'Data Science'
+  - 'AI'
+Tags:
+  - 'Search Algorithms'
```
[the remark this answers is not in the document]
Comment (HishamT) [reply content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md:8]: Done!

**Review by KTom101 — APPROVED** [review]
@HishamT This can go on to the next review. But the exclamation points for the linked images on lines 36-59 weren't supposed to be removed since they were images. To link an image, the syntax would be:
 `name`

Regular links to webpages would be `[text](link)` which doesn't have the exclamation point. So the only exclamation points that needed to be removed were the ones that were linking out other entries which thank you for doing 😄 Hope my exclamation makes sense!

File: content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+7. Select the top β nodes and add them to the tree, connecting them to their respective parent nodes.
+8. Repeat this process until the goal node **G** is removed from the open list, indicating that a path has been found.
+
+It is important to note that every level of the search tree contains at most β number of nodes.
+
+## Example
+
+Consider the following example of searching a graph from **S** to **G**. For simplicity, only the edge weights will be used for evaluation. A beam width of β = 2 will be used, meaning only the top two nodes are considered at every level.
+
+[Example-graph](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-example-graph.png)
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md:36]:
```suggestion
![Example-graph](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-example-graph.png)
```

File: content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+Initialize the tree and add **S** to the **closed list**:
+
+[BEAM-Search-1](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-Tree-1.png)
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md:40]:
```suggestion
![BEAM-Search-1](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-Tree-1.png)
```

File: content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+**S** only has one adjacent node, so it is added to the tree:
+
+[BEAM-Search-2](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-Tree-2.png)
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md:44]:
```suggestion
![BEAM-Search-2](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-Tree-2.png)
```

File: content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+Explore **I**. Nodes **N** and **H** are the top two nodes, so those are added to the tree:
+[BEAM-Search-3](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-Tree-3.png)
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md:47]:
```suggestion
![BEAM-Search-3](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-Tree-3.png)
```

File: content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+Nodes **F** (adjacent to **H** in the graph) and **L** (adjacent to **N** in the graph) are next:
+[BEAM-Search-4](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-Tree-4.png)
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md:50]:
```suggestion
![BEAM-Search-4](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-Tree-4.png)
```

File: content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+The incompleteness of this algorithm can be seen in this step. Node **G** has appeared in the **open list** but unfortunately, it has been eliminated by the beam width. Nodes **M** and **C** are next:
+[BEAM-Search-5](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-Tree-5.png)
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md:53]:
```suggestion
![BEAM-Search-5](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-Tree-5.png)
```

File: content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+It is in this step that node **G** has been found:
+[BEAM-Search-6](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-Tree-6.png)
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md:56]:
```suggestion
![BEAM-Search-6](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-Tree-6.png)
```

File: content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+Trace the path from **S** to **G**:
+[Final-path](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-final-path.png)
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md:59]:
```suggestion
![Final-path](https://raw.githubusercontent.com/Codecademy/docs/main/media/BEAM-Search-final-path.png)
```

File: content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
+---
+Title: 'BEAM Search'
+Description: 'A graph searching algorithm that is a variant of breadth-first search'
```
Comment (SSwiniarski) [inline content/ai/concepts/search-algorithms/terms/beam-search/beam-search.md:3]:
```suggestion
Description: 'A graph searching algorithm that is a variant of breadth-first search.'
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@HishamT I did a second review. My comments are just fixing the image markdown and adding a period.

**Review by SSwiniarski — APPROVED** [review]
@HishamT Looks good to me. Shall we merge?

**Comment by HishamT** [comment]
Sounds good! :)

**Pull request merged**